File: Algorithm/Python/1000+/1535. Find the Winner of an Array Game.py
class Solution(object):
    def getWinner(self, arr, k):
        """
        :type arr: List[int]
        :type k: int
        :rtype: int
        """
        idx, cnt, curr_num = 1, 0, arr[0]
        while idx < len(arr):
            if curr_num < arr[idx]:
                curr_num = arr[idx]
                cnt = 1
            else:
                cnt += 1
            if cnt == k:
                break
            idx += 1
        return curr_num

File: Algorithm/C++/1000+/1535. Find the Winner of an Array Game.cc
#include <vector>

class Solution {
public:
    int getWinner(std::vector<int>& arr, int k) {
        int cnt = 0, curr_num = arr[0];
        for(int i=1; i<arr.size(); i++) {
            if(curr_num < arr[i]) {
                cnt = 1;
                curr_num = arr[i];
            } else {
                cnt++;
            }
            if (cnt == k) break;
        }
        return curr_num;
    }
};